api/src/websocket/controllers/rest.ts

```diff
@@ -0,0 +1,55 @@
+import type WebSocket from 'ws';
+import type { Server as httpServer } from 'http';
+import type { AuthenticationState, AuthMode, WebSocketClient } from '../types';
+import env from '../../env';
+import SocketController from './base';
+import emitter from '../../emitter';
+import { refreshAccountability } from '../authenticate';
+import { handleWebsocketException, WebSocketException } from '../exceptions';
+import logger from '../../logger';
+import { WebSocketMessage } from '../messages';
+import { parseJSON } from '@directus/shared/utils';
+
+export class WebsocketController extends SocketController {
+	constructor(httpServer: httpServer) {
+		super(httpServer, 'WS REST', String(env['WEBSOCKETS_REST_PATH']), {
+			mode: String(env['WEBSOCKETS_REST_AUTH']).toLowerCase() as AuthMode,
+			timeout: Number(env['WEBSOCKETS_REST_AUTH_TIMEOUT']) * 1000,
```

Comment:
shall we move the logic of the above two lines into the abstract class and then just pass the names of the environment variables to it? this would remove some duplicated code like the casting, the lowercase function call and the multiplication within is currently done in both concrete classes 🤔

Reply:
actually also for the logger statement at the end of the constructor. if you would move this statement out of the concrete class and use the `name` variable in the abstract class, you could even get rid of the dependency to `../../env` in the concrete classes

Reply:
Hmmm interesting suggestion 🤔 We could probably, abstract this into a "getConfig" function but im not 💯% sure that it's worth the abstraction for these 2 lines.
This was originally setup with the idea that these configs could diverge at some point but that never happened, will definitely have a quick look if a sensible abstraction can be done without too much hassle 😄

Reply:
compared to what you all did in this PR this shouldn't be too much hassle 😆😆  very nice work by the way!!